src/app/public/modules/phone-field/phone-field.component.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  OnDestroy
+} from '@angular/core';
+
+require('intl-tel-input/build/js/utils');
+require('intl-tel-input/build/js/intlTelInput');
+
+/**
+ * NOTE: We can not type these due the the current @types/intl-tel-input version having an
+ * undeclared type which causes linting errors.
+ */
+declare var intlTelInputUtils: any;
+declare var intlTelInputGlobals: any;
+
+import {
+  SkyCountryData
+} from './types';
+
+@Component({
+  selector: 'sky-phone-field',
+  templateUrl: './phone-field.component.html',
+  styleUrls: ['./phone-field.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyPhoneFieldComponent implements OnDestroy {
+
+  public countryData: SkyCountryData[];
+
+  public defaultCountryData: SkyCountryData;
+
+  public disabled: boolean;
+
+  public set selectedCountry(newCountry: SkyCountryData) {
+    if (this._selectedCountry !== newCountry) {
+      this._selectedCountry = newCountry;
+
+      if (!this._selectedCountry.placeholder) {
+        this._selectedCountry.placeholder = intlTelInputUtils.getExampleNumber(
+          newCountry.iso2,
+          true,
+          intlTelInputUtils.numberType.FIXED_LINE
+        );
+      }
+
+      this.sortCountriesWithSelectedAndDefault(newCountry);
+
+      this.selectedCountryChange.emit(newCountry);
+    }
+  }
+
+  public get selectedCountry() {
+    return this._selectedCountry;
+  }
+
+  public selectedCountryChange = new EventEmitter<SkyCountryData>();
+
+  private _selectedCountry: SkyCountryData;
+
+  constructor() {
+    /**
+     * The "slice" here ensures that we get a copy of the array and not the global original. This
+     * ensures that multiple instances of the component don't overwrite the original data.
+     */
+    this.countryData = intlTelInputGlobals.getCountryData().slice(0);
+    this.selectedCountry = this.countryData[0];
+  }
+
+  public ngOnDestroy(): void {
+    this.selectedCountryChange.complete();
+  }
+
+  /**
+   * Format's the given phone number based on the currently selected country.
+   * @param phoneNumber The number to format
+   */
+  public formatNumber(phoneNumber: string): string {
```

Comment:
There are a few `public` methods here that are only being used by the directive. What's keeping us from putting these in the directive and making them `private`?

Reply:
Researching why I did that. I don't remember for sure.

Reply:
I have made some API changes based on offline discussion that cleaned this up.